test/unit/utils.test.ts

```diff
@@ -0,0 +1,199 @@
+			assert.ok(result, 'getHex should return a value')
+			assert.ok(Array.isArray(result), 'Result should be an array')
+			assert.strictEqual(
+				result.length,
+				2,
+				'Result should have 2 elements',
+			)
+			assert.ok(
+				result[0] >= 0 && result[0] <= 255,
+				'First byte should be between 0 and 255',
+			)
+			assert.ok(
+				result[1] >= 0 && result[1] <= 255,
+				'Second byte should be between 0 and 255',
+			)
+		})
+
+		it('should handle negative numbers', () => {
+			const result = getHex(-12.34)
+			assert.ok(
+				result,
+				'getHex should return a value for negative numbers',
+			)
+			assert.ok(Array.isArray(result), 'Result should be an array')
+			assert.strictEqual(
+				result.length,
+				2,
+				'Result should have 2 elements',
+			)
+			assert.ok(
+				result[0] & 0x80,
+				'Sign bit should be set for negative numbers',
+			)
+		})
+
+		it('should handle zero', () => {
+			const result = getHex(0)
+			assert.ok(result, 'getHex should return a value for zero')
+			assert.deepStrictEqual(
+				result,
+				[0, 0],
+				'Result for zero should be [0, 0]',
+			)
+		})
+
+		it('should handle very large numbers', () => {
+			const result = getHex(1e38)
+			assert.ok(
+				result,
+				'getHex should return a value for very large numbers',
+			)
+			assert.ok(Array.isArray(result), 'Result should be an array')
+			assert.strictEqual(
+				result.length,
+				2,
+				'Result should have 2 elements',
+			)
+		})
+
+		it('should handle very small numbers', () => {
+			const result = getHex(1e-38)
+			assert.ok(
+				result,
+				'getHex should return a value for very small numbers',
+			)
+			assert.ok(Array.isArray(result), 'Result should be an array')
+			assert.strictEqual(
+				result.length,
+				2,
+				'Result should have 2 elements',
+			)
+		})
+	})
+
+	describe('getFloat', () => {
+		it('should convert 2-byte array to float', () => {
+			const hex = getHex(12.34)
+			assert.ok(hex)
+			const float = getFloat(hex[0], hex[1])
+			assert.ok(Math.abs(float - 12.34) < 0.01)
+		})
+
+		it('should handle negative numbers', () => {
+			const hex = getHex(-12.34)
+			assert.ok(hex)
+			const float = getFloat(hex[0], hex[1])
+			assert.ok(Math.abs(float + 12.34) < 0.01)
+		})
+	})
+
+	describe('wait', () => {
+		it('should wait for the specified time', async () => {
+			const start = Date.now()
+			await wait(100) // wait for 100 ms
+			const elapsed = Date.now() - start
+			assert(elapsed >= 100 && elapsed < 150) // Allow some margin for execution time
+		})
+	})
+
+	describe('round', () => {
+		it('should round numbers correctly', () => {
+			assert.strictEqual(round(1.2345, 2), 1.23)
+			assert.strictEqual(round(1.2365, 2), 1.24)
+			assert.strictEqual(round(123.456, 0), 123)
+		})
+
+		it('should handle negative numbers', () => {
+			assert.strictEqual(round(-1.2345, 2), -1.23)
+			assert.strictEqual(round(-1.2365, 2), -1.24)
+		})
+
+		it('should handle zero', () => {
+			assert.strictEqual(round(0, 2), 0)
+		})
```

Comment:
add a case here that covers possible js round errors. See https://www.robinwieruch.de/javascript-rounding-errors/

Reply:
Done, used Number.EPSILON